From ea5719a712c39bcb20d3b4c557a76d4443e96dfa Mon Sep 17 00:00:00 2001
From: Abhijit Ayarekar <aayarekar@marvell.com>
Date: Wed, 12 Aug 2020 12:01:25 -0700
Subject: [PATCH 07/19] octeontx2-dpi: Export get_vf_count api.

commit 9e85fcabc65149a51bf377734847cbd2971cf8d8 from
git@git.assembla.com:cavium/WindRiver.linux.git

Kernel drivers can use this api to get number of configured
vf's.

Change-Id: Ie4ee355fc3e7ef649bf3faa15f37c5fabfea61b5
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/33901
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49030
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-dpi/dpi.c | 10 +++++++++-
 drivers/soc/marvell/octeontx2-dpi/dpi.h |  1 +
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2-dpi/dpi.c b/drivers/soc/marvell/octeontx2-dpi/dpi.c
index 78b52004eda8..0cdeb43a26f0 100644
--- a/drivers/soc/marvell/octeontx2-dpi/dpi.c
+++ b/drivers/soc/marvell/octeontx2-dpi/dpi.c
@@ -461,8 +461,16 @@ static int dpi_queue_config(struct pci_dev *pfdev,
 	return queue_config(dpi, dpivf, msg);
 }
 
+static int dpi_get_vf_count(struct pci_dev *pfdev)
+{
+	struct dpipf *dpi = pci_get_drvdata(pfdev);
+
+	return dpi->total_vfs;
+}
+
 struct otx2_dpipf_com_s otx2_dpipf_com  = {
-	.queue_config = dpi_queue_config
+	.queue_config = dpi_queue_config,
+	.get_vf_count = dpi_get_vf_count
 };
 EXPORT_SYMBOL(otx2_dpipf_com);
 
diff --git a/drivers/soc/marvell/octeontx2-dpi/dpi.h b/drivers/soc/marvell/octeontx2-dpi/dpi.h
index 6a359fe25ecc..e35478610ed0 100644
--- a/drivers/soc/marvell/octeontx2-dpi/dpi.h
+++ b/drivers/soc/marvell/octeontx2-dpi/dpi.h
@@ -256,6 +256,7 @@ union dpi_mbox_message_t {
 struct otx2_dpipf_com_s {
 	int (*queue_config)(struct pci_dev *pfdev,
 			    union dpi_mbox_message_t *req);
+	int (*get_vf_count)(struct pci_dev *pfdev);
 };
 
 extern struct otx2_dpipf_com_s otx2_dpipf_com;
-- 
2.31.1

